### ISSUE TITLE
React 15.2 has bunch of unknown props warnings

### ISSUE THREAD

**Issue by tquetano-r7** [zippyui/react-date-picker#117]
With v15.2 of React, they started throwing warnings on any standard HTML elements that were passed invalid / unknown props (ones that do not translate to valid HTML attributes). As such, we are now flooded with a ton of warnings, example:

```
warning.js?8a56:44Warning: Unknown prop `flex` on <div> tag. Remove this prop from the element. For details, see https://fb.me/react-unknown-prop
    in div (created by FlexItem)
    in FlexItem (created by NavBar)
    in div (created by Flex)
    in Flex (created by NavBar)
    in NavBar (created by BasicMonthView)
    in div (created by Flex)
    in Flex (created by BasicMonthView)
    in BasicMonthView (created by MonthView)
    in MonthView (created by CalendarWithTime)
    in div
    in CalendarWithTime
    in div
    in div (created by TabPanel)
    in TabPanel
    in div (created by Tabs)
    in Tabs
    in div
    in SelectTimeRange (created by PageBanner)
```

Calling it out as they should probably be cleaned up.

**Comment by radubrehar** [comment]
Yep, working on updating the picker to fix these changes. Expect a release probably tomorrow.

**Comment by tquetano-r7** [comment]
Sweet! thanks

**Comment by radubrehar** [comment]
`5.3.17` is out on npm with these fixes. I'm not sure we cleaned-up all uknown props, but we'll have some rounds of testing and fix other props as well as they come up.

**Comment by af7** [comment]
URGENT: I get this problem (and the application seems to crash) after upgrading:

![image](https://cloud.githubusercontent.com/assets/7850073/16686008/a4a20f6a-4517-11e6-9cdf-1ae078240f2f.png)

What can we do about this?

**Comment by pointblack** [comment]
got the same issue

**Comment by radubrehar** [comment]
A snippet of code would be useful so we can repro

**Comment by JohnyDays** [comment]
I'm also getting the same error, I just installed react-date-picker and imported it, and react appears to be loading from a production version and then crashing.

**Comment by JohnyDays** [comment]
@radubrehar I'm available for debugging if you need something. The component looks awesome and I wanted to use it 😭

**Comment by radubrehar** [comment]
Yes, we're currently investigating the issue. @JohnyDays

**Comment by radubrehar** [comment]
@JohnyDays @pointblack @af7 you should be all fine now with `5.3.18`
It's been a problem with one of our dependencies. Fixed it and made sure we use exact version instead of caret semver.

**Comment by pointblack** [comment]
great thanks, works with 5.3.18 👍

**Comment by JohnyDays** [comment]
Works perfectly, thanks 📆

**Comment by af7** [comment]
In my case there is this problem:
![image](https://cloud.githubusercontent.com/assets/7850073/16694804/dc40739e-4544-11e6-9450-1930f48da717.png)

Note that I reverted to React 15.1.0 so that I do not have the warnings about props, etc. (on other packages)

**Comment by dkh215** [comment]
I'm getting the same error as @af7:

<img width="1267" alt="screen shot 2016-07-08 at 4 14 40 pm" src="https://cloud.githubusercontent.com/assets/16489251/16700266/2aa0432a-4527-11e6-9bf0-3d011f608004.png">

<img width="416" alt="screen shot 2016-07-08 at 4 14 09 pm" src="https://cloud.githubusercontent.com/assets/16489251/16700272/3036e0c8-4527-11e6-9a7b-044b5acc76ea.png">

**Comment by JohnyDays** [comment]
forceValidDate causes this error, removing it seems to solve the problem

**Comment by radubrehar** [comment]
We've shiped `5.3.19` with the fixes.
The problem is we were in a rush to remove unknown props for compat with React `15.2` and we deleted too many props. Sorry for that.
After the event we've decided that we go for automated tests, so next week we're starting to write tests and integrate with CI to ensure code quality.

**Comment by af7** [comment]
Thank you!

**Comment by dkh215** [comment]
@JohnyDays thanks!

**Comment by dev-zippyui** [comment]
Has anyone spotted any other warnings for unknown props?

**Comment by vnikolova** [comment]
I get quite a lot of these: 
![screenshot 2](https://cloud.githubusercontent.com/assets/6525494/17214231/2e0112ae-54d8-11e6-872e-fbb9d8dc46c2.png)

**Comment by Inlesco** [comment]
React 15.2.0:
![screenshot_7](https://cloud.githubusercontent.com/assets/5101235/17288448/e13518ae-57dd-11e6-8a61-097db24a1186.png)

Setup: 
![screenshot_8](https://cloud.githubusercontent.com/assets/5101235/17288454/e7d591f2-57dd-11e6-9ec3-b18551858123.png)

**Comment by bdwain** [comment]
i still get these

```
warning.js:25 Warning: Unknown props `defaultRange`, `onRangeChange`, `rangeStart` on <div> tag. Remove these props from the element. For details, see https://fb.me/react-unknown-prop
    in div (created by Flex)
    in Flex (created by BasicMonthView)
    in BasicMonthView (created by MonthView)
    in MonthView (created by RangePickerPopover)
```

**Comment by dev-zippyui** [comment]
`5.3.27` fixes those listed above

**Comment by bdwain** [comment]
awesome! thanks!

**Comment by erisnuts** [comment]
I have several warnings about unknown props in MultiMounthView:
![screenshot-15](https://cloud.githubusercontent.com/assets/3906376/17477975/02b547de-5d82-11e6-9763-57eabc9e8744.png)
![screenshot-14](https://cloud.githubusercontent.com/assets/3906376/17477976/02b58c4e-5d82-11e6-9131-4d2f7502eedd.png)
![screenshot-13](https://cloud.githubusercontent.com/assets/3906376/17477977/02b6634e-5d82-11e6-9f94-fc14da272187.png)
![screenshot-12](https://cloud.githubusercontent.com/assets/3906376/17477978/02b7b276-5d82-11e6-857d-b55f578f375c.png)

**Comment by markploetneretecture** [comment]
I have several warnings too with v5.3.27
+
![datepicker-warnings](https://cloud.githubusercontent.com/assets/13830952/17587742/5fd0cb56-5fca-11e6-924c-67d7079286bf.png)

**Comment by dev-zippyui** [comment]
Just released `5.3.28` to npm - see https://github.com/zippyui/react-date-picker/releases/tag/v5.3.28 with fixes for these.

You might also want to have a look at the new live sample snippets in the examples site at http://zippyui.com/docs/react-date-picker/examples/1

**Comment by erisnuts** [comment]
After update to v5.3.28 still have two warnings:

![screenshot-34](https://cloud.githubusercontent.com/assets/3906376/17734894/966f4926-6498-11e6-8096-b75720b20156.png)

![screenshot-35](https://cloud.githubusercontent.com/assets/3906376/17734896/9851d678-6498-11e6-9089-955ab605a12b.png)

(React version 15.2.1)

**Comment by TigerC10** [comment]
If you adopt a higher stage for ES2015, then you can put a `static propTypes` var directly in the class.  Then, later, when you're passing props down to child elements you can use a pattern like this:

```javascript
  _filterProps(props) {
    return _.omit(props, [
      ...Object.keys(ParentComponent.propTypes),
      ...Object.keys(ParentComponent.defaultProps),
    ]);
  }

  render() {
    return (
      <ChildComponent ...this._filterProps(this.props) />
    );
  }
```

Point is, rather than blacklisting props with delete calls (the current solution) you can whitelist props that pass down to the child by passing them in the child prop attributes.

**Comment by micheleb** [comment]
using a `DatePicker`, version `5.3.28` (with React `15.3.1`):

    Warning: Unknown props `cancelButton`, `todayButton`, `clearButton` on <div> tag. 
    Remove these props from the element. For details, see https://fb.me/react-unknown-prop
        in div (created by Flex)                                                
        in Flex (created by BasicMonthView)                                     
        in BasicMonthView (created by MonthView)                                
        in MonthView (created by Calendar)                                      
        in div (created by Flex)                                                
        in Flex (created by Calendar)                                           
        in Calendar (created by DateField)

**Comment by dev-zippyui** [comment]
@TigerC10 this is the solution we're going to implement - whitelisting props. A new release will fix this.

**Comment by erisnuts** [comment]
![screenshot-59](https://cloud.githubusercontent.com/assets/3906376/18344866/763c48e8-75d2-11e6-9ed6-5297bea0d99b.png)
![screenshot-58](https://cloud.githubusercontent.com/assets/3906376/18344865/763c3560-75d2-11e6-9c4c-0c88c78b5d4a.png)

Last one - it's something new, didn't see it earlier.
I hope it helps somehow.)

**Comment by jrnail23** [comment]
any movement on this issue?  It's proving to be quite a headache for my team.
